src/services/bitbucket/BitbucketService.ts

```diff
@@ -251,14 +253,44 @@ export class BitbucketService implements IVCSService {
     return { items, ...pagination };
   }
 
-  async listIssues(owner: string, repo: string): Promise<Paginated<Issue>> {
+  async listIssues(
+    owner: string,
+    repo: string,
+    options?: { withDiffStat?: boolean } & ListGetterOptions<{ state?: IssueState }>,
+  ): Promise<Paginated<Issue>> {
     this.authenticate();
+    const apiUrl = `https://api.bitbucket.org/2.0/repositories/${owner}/${repo}/issues`;
+
+    const state = VCSServicesUtils.getIssueState(options?.filter?.state, VCSServiceType.bitbucket);
+    // put state in quotation marks because of Bitbucket API https://developer.atlassian.com/bitbucket/api/2/reference/meta/filtering#query-issues
+    let quotedState: string | string[] = `"${state}"`;
+    if (_.isArray(state)) {
+      quotedState = state.map((state) => {
+        return `"${state}"`;
+      });
+    }
 
-    const params: Bitbucket.Params.IssueTrackerList = {
-      repo_slug: repo,
-      username: owner,
+    // get q parameter
+    const stringifiedState = qs.stringify(
+      { state: quotedState },
+      {
+        addQueryPrefix: false,
+        encode: false,
+        arrayFormat: 'repeat',
+        delimiter: '+OR+',
+      },
+    );
```

Comment:
2 times the same code - create some util function and add tests for it.

Reply:
Is it possible to optimize the logic?

```ts
    let quotedState: string | string[] = `"${state}"`;
    if (_.isArray(state)) {
      quotedState = state.map((state) => {
        return `"${state}"`;
      });
    }
```
to:
```ts
    const quotedState: state.map((state) => {
        return `"${state}"`;
      });
```
It seems that it's not necessary to split logic for string and array.

In the best:
```ts
    const quotedState: state.map((state) =>`"${state}"`);
```

Reply:
It's not possible to call `.map()` on string

Reply:
ok, good point